realm/src/main/java/io/realm/RealmBaseAdapter.java

```diff
@@ -89,7 +94,20 @@ public long getItemId(int i) {
      * @param realmResults the new RealmResults coming from the new query.
      */
     public void updateRealmResults(RealmResults<T> realmResults) {
+        if (listener != null) {
+            if (this.realmResults != null) {
+                this.realmResults.getRealm().removeChangeListener(listener);
+            }
+            if (realmResults != null) {
+                realmResults.getRealm().addChangeListener(listener);
+            }
+        }
+
         this.realmResults = realmResults;
-        notifyDataSetChanged();
+        if (this.realmResults == null) {
+            notifyDataSetInvalidated();
```

Comment:
I am not sure that calling this here would be correct. JavaDoc says that once this has been called the adapter should not be used anymore, which is not correct in this case, it could become "valid" again by just updating the results agains.